plugin/ensime.vim

```diff
@@ -12,6 +12,7 @@ augroup END
 
 command! -nargs=* -range EnNoTeardown call ensime#com_en_no_teardown([<f-args>], '')
 command! -nargs=* -range EnTypeCheck call ensime#com_en_type_check([<f-args>], '')
+command! -nargs=* -range EnBufferedTypeCheck call ensime#com_en_buf_type_check([<f-args>], '')
```

Comment:
I know this is rough-ish implementation, but just asking for sake of design discussion: does this really need to exist as a separate command?

If a bit of testing proves this to be robust and trouble-free, it seems entirely reasonable to me for this to be in the default code path for `:EnTypeCheck`. If it's potentially controversial, then *maybe* have an option to disable the buffering, but I lean toward not having tons of options until/unless someone makes a good case for wanting them.